frontend/src/utils/Utils.ts

```diff
@@ -21,3 +23,28 @@ export const embedLink = (originalYoutubeLink: string): string => {
 
   return originalYoutubeLink.replace("watch?v=", "embed/");
 };
+
+export const insertSortedComments = (
+  existingComments: CommentResponse[],
+  newComment: CommentResponse,
+): CommentResponse[] => {
```

Comment:
nit: something like this might look a little cleaner for the loop
```
    if (existingComments[i].lineIndex > newComment.lineIndex) break;
    i += 1;
    if (
      existingComments[i].lineIndex === newComment.lineIndex &&
      existingComments[i].commentIndex === newComment.commentIndex - 1
    ) {
      break;
    }
```
either this or the original code look good to me so ill leave it up to you!